lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -3567,13 +3565,18 @@ bool LLVMToSPIRVBase::isAnyFunctionReachableFromFunction(
   return false;
 }
 
-void LLVMToSPIRVBase::collectInputOutputVariables(SPIRVFunction *SF,
-                                                  Function *F) {
+std::vector<SPIRVId>
+LLVMToSPIRVBase::collectEntryPointInterfaces(SPIRVFunction *SF, Function *F) {
+  std::vector<SPIRVId> Interface;
   for (auto &GV : M->globals()) {
-    const auto AS = GV.getAddressSpace();
-    if (AS != SPIRAS_Input && AS != SPIRAS_Output)
-      continue;
-
+    SPIRVModule *BM = SF->getModule();
+    if (!BM->isAllowedToUseVersion(VersionNumber::SPIRV_1_4)) {
+      const auto AS = GV.getAddressSpace();
+      if (AS != SPIRAS_Input && AS != SPIRAS_Output)
+        continue;
+    } else {
+      BM->setSPIRVVersion(static_cast<SPIRVWord>(VersionNumber::SPIRV_1_4));
```

Comment:
```suggestion
      BM->setMinSPIRVVersion(static_cast<SPIRVWord>(VersionNumber::SPIRV_1_4));
```

Reply:
It seems like by default(unless `--spirv-max-version` option is used) we will generate SPIR-V version 1.4 every time there is a global variable used by a kernel, i.e. almost always. In case there are concerns about that, probably it makes sense to create a separate PR which will rise the version based on features in the module(not only for this case).

Reply:
We can rise version if translator allowed to use v1.4 & interface variables are not of input/output storage class. It should reduce the number of such cases.

Reply:
0462754e8b1fe484a29ce21bf0d622eda7bc6460

Reply:
Applied in two last commits